### ISSUE TITLE
Support for opening links by default

### ISSUE THREAD

**Issue by MulverineX** [soundcrowd/soundcrowd-plugin-soundcloud#3]
Manually enabled Open by Default support for
- `soundcloud.com`
- `m.soundcloud.com`
- `on.soundcloud.com`

Irrelevant sidenote: how did you manage to get an API key for this, and if you didn't do anything special, do you have a back up plan for when its inevitably revoked? Related Art: https://github.com/faridrasidov/soundcld https://github.com/njasm/KotWave https://github.com/TeamNewPipe/NewPipeExtractor

**Comment by 0xf4b1** [comment]
I'll look into that, however I believe the domains must be defined in the main app which is not dynamic based on which plugins are installed but maybe I can make an exception for soundcloud.

Regarding API key, I believe you could do everything without one by logging in via the web page and reading the token from cookies as other projects do it.

**Comment by MulverineX** [comment]
> I'll look into that, however I believe the domains must be defined in the main app which is not dynamic based on which plugins are installed but maybe I can make an exception for SoundCloud.

Any android app can have Open by Default configured, even if it doesn't have a UI, afaik. I may be wrong though. If it does require a UI to be defined it can just be a dummy UI.

> Regarding API key, I believe you could do everything without one by logging in via the web page and reading the token from cookies as other projects do it.

The problem is that the user token won't work to access the 3rd-party developer API; it's only scoped for their closed 1st-party API.

**Comment by 0xf4b1** [comment]
Yes, you are probably right with that. But the plugins actually never run itself and just contain the code that is loaded by the main app. It's maybe possible to let the plugin run the main app and pass the url. But what I actually had in mind would be directly opening the main app and delegating the handling of the url to the matching plugin.

Regarding the API, yes I remember also noticing that the token only works with the api-v2 when I was experimenting to get some data but I think its very similar.
